catalyst/dl/callbacks/mixup.py

```diff
@@ -1,50 +1,56 @@
 from typing import List
-from catalyst.dl.core.state import RunnerState
-from catalyst.dl.callbacks import CriterionCallback
+
 import numpy as np
+
 import torch
 
+from catalyst.dl.core.state import RunnerState
+from catalyst.dl.callbacks import CriterionCallback
+
 
 class MixupCallback(CriterionCallback):
-    def __init__(self, fields: List[str] = ('features',), alpha=1.0,
+    def __init__(self, fields: List[str] = ("features",), alpha=1.0,
                  train_only=True, **kwargs):
-        assert len(
-            fields) > 0, 'At least one field for MixupCallback is required'
-        assert alpha >= 0, 'alpha must be>=0'
+        assert len(fields) > 0, \
+            "At least one field for MixupCallback is required"
+        assert alpha >= 0, "alpha must be>=0"
 
-        super(MixupCallback, self).__init__(**kwargs)
+        super().__init__(**kwargs)
 
         self.train_only = train_only
         self.fields = fields
         self.alpha = alpha
         self.lam = 1
         self.index = None
 
+    def _is_needed(self, state: RunnerState):
+        return not self.train_only or state.loader_name.startswith("train")
```

Comment:
what about check this flag `on_loader_start`?

Reply:
Thanks, done.